.github/workflows/test.yml

```diff
@@ -69,119 +65,108 @@ jobs:
+          context: .
+          file: ./cmd/shfmt/Dockerfile
+          platforms: ${{ env.DOCKER_PLATFORMS }}
+          push: true
+          tags: ${{ env.DOCKER_BASE }}:${{ env.TAG }}
+      - name: Build and push (alpine)
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./cmd/shfmt/Dockerfile
+          platforms: ${{ env.DOCKER_PLATFORMS }}
+          push: true
+          tags: ${{ env.DOCKER_BASE }}:${{ env.TAG }}-alpine
+          target: alpine
+      - name: Test multi-architecture Docker images
+        run: |
+          # Enable docker daemon experimental support (for 'pull --platform').
+          config='/etc/docker/daemon.json'
+          if [[ -e "$config" ]]; then
+            sudo sed -i -e 's/{/{ "experimental": true, /' "$config"
+          else
+            echo '{ "experimental": true }' | sudo tee "$config"
+          fi
+          sudo systemctl restart docker
+          printf '%s\n' "#!/bin/sh" "echo 'hello world'" >myscript
+          for platform in ${DOCKER_PLATFORMS//,/ }; do
+            for ext in '' '-alpine'; do
+              image="${DOCKER_BASE}:${TAG}${ext}"
+              msg="Testing docker image $image on platform $platform"
+              line="${msg//?/=}"
+              printf "\n${line}\n${msg}\n${line}\n"
+              docker pull -q --platform "$platform" "$image"
+              if [ -n "$ext" ]; then
+                echo -n "Image architecture: "
+                docker run --rm --entrypoint /bin/sh "$image" -c 'uname -m'
+              fi
+              version=$(docker run --rm "$image" --version)
+              echo "shfmt version: $version"
+              if [ "$TAG" != 'latest' -a "$TAG" != "$version" ]; then
+                echo "Version mismatch: shfmt $version tagged as $TAG"
+                exit 1
+              fi
+              docker run --rm -v "$PWD:/mnt" -w '/mnt' "$image" -d myscript
+            done
           done
-        done
-    - name: Install Node.js
-      uses: actions/setup-node@v1
-      with:
-        node-version: 13.x
-    - name: Update DockerHub description
-      run: |
-        set -vx
-        npm install docker-hub-api@0.8.0
-        node -e '
-          function error(reason) {
-            console.log("Error: " + reason.message);
-            process.exit(1);
-          }
-          const fs = require("fs");
-          let readme = fs.readFileSync("README.md", "utf8");
-          let dockerHubAPI = require("docker-hub-api");
-          dockerHubAPI.login(
-            process.env.DOCKER_USER,
-            process.env.DOCKER_PASSWORD)
-          .then(function () {
-            let url = "https://github.com/" + process.env.GITHUB_REPOSITORY;
-            let repo = process.env.DOCKER_REPO;
-            dockerHubAPI.setRepositoryDescription(
+      - name: Install Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 13.x
+      - name: Update DockerHub description
```

Comment:
You might also want to replace this hand-rolled code with https://github.com/peter-evans/dockerhub-description

Reply:
Yes, fully agree, that's why I also suggested it in above ;-)

To be honest, I do not like so much the adherence to GitHub that this PR brings and you code was very instructive and useful to understand the workflow! I'll definitively take a look at your post.

Anyway, in the long run, this PR will help maintenance so I guess it's a good step forward :-)